### DISCUSSION TITLE
ACAP support for Yolov5 object detector model

### DISCUSSION THREAD

**Discussion by Mahdi89** [AxisCommunications/discussions#60] [ACAP] [answered]
Hi there,

I would like to know if there is ACAP v4 support for the latest object detectors e.g. Yolov5? The default seems to be mobilenet v2 which is not as perfromant as Yolo architecture. Looking forward to hearing from you.

Best
MJ

**Comment by Mahdi89** [comment, marked as answer]
To follow us, Tensorflow edge TPU model is available so I wonder if it'd be possible to run the model using ACAP SDK?
https://zenodo.org/record/6222936#.YySV_XXMJH5

Thanks

**Reply by Corallo** [reply]
Hello, thanks for your post, it was interesting to see that yolov5 supports now edgetpu.
I am having errors when I try to run the export script in the yolov5,
but I tried to run the 224x224 version of the small-size yolov5 that is pre-converted to edgetpu in this repository:
https://github.com/jveitchmichaelis/edgetpu-yolo
It seems to work on artpec7 edgetpu, with an inference time of 30-40 ms
Didn't check the quality of the detections tho :)

**Reply by Mahdi89** [reply]
Thanks so much @Corallo, the small model's mAP is little too low  but it should suffice for now. I'll let you know about the performnace of the converted model. Thanks again.

**Reply by Corallo** [reply]
Sounds great! Looking forward to hearing about your results!

**Reply by Abiganesh** [reply]
Hi, are you guy able to find a way to run yolo v8 or v5 on ACAP.

**Reply by Corallo** [reply]
@Abiganesh 
For what we saw experimenting with Yolov5/8, it is not suitable to run on an ACAP. It is possible in some cases, but the drawbacks make it not a good choice. An SSD-mobilenet 640x640 would perform better.

**Reply by Abiganesh** [reply]
I would like to attempt running YOLOv8 on an ACAP. Can you assist me in understanding how to do so?

**Reply by Corallo** [reply]
You can look here at attempts made by other users:

https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/discussions?discussions_q=is%3Aopen+yolo
https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/issues?q=yolo

Here there are more general guidelines
https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/discussions/112 

If you want to try anyway, running yolo on DLPU is going by hard.
Your objective is to obtain a TFlite model, **quantized by tensor**

If you are fine running it on CPU, 
it might be sufficient to export yolov8 on TFlite using their export tool, and replace the model in one of these two examples:
https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/tree/main/minimal-ml-inference
https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/tree/main/object-detector-python

**Reply by kondaredddy** [reply]
Is there a full tutorial for implementing a custom SSD-MobileNet model on ACAP?

**Reply by Corallo** [reply]
@kondaredddy There is not an end-to-end guide, but take a look at this answer,
https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/discussions/169
It had some links to a guide to train and one to convert that you could follow

**Comment by Corallo** [comment]
Hi, 
It is now possible to run Yolov5 on Artpec-8 on devices running Axis OS >= 11.7
See the guide [here](https://github.com/AxisCommunications/axis-model-zoo/blob/main/docs/yolov5-on-artpec8.md)
